### ISSUE TITLE
feat: Connecting snowflake using Key Pair Authentication

### ISSUE THREAD

**Issue by uramith** [ibis-project/ibis#9837]
### Is your feature request related to a problem?

Hi All,

I have been trying to connect to snowflake data base using encrypted key, but no luck so far. I couldn't find a straight forward way of including private key while connecting to snowflake using ibis `connect()`. Just wondering, how to make connection using **private_key**. If this feature is not present already, please consider this feature in the upcoming release. 

I am able to connect to db using below method
``` 
import snowflake.connector

#Create a Snowflake connection object
conn = snowflake.connector.connect(
    user=SNOWFLAKE_USERNAME,
    account=SNOWFLAKE_ACCOUNT,
    private_key=SNOWFLAKE_PKEY,    # extracted private key from .p8 file
    warehouse=WAREHOUSE,
    database=DATABASE,
    schema=SCHEMA  
)

#Create a cursor object
cur = conn.cursor()
```

After surfing the web, I got a post where they made connection using below methods. I have tried that method using sqlalchemy, but no luck.
```
from sqlalchemy import create_engine
from snowflake.sqlalchemy import URL

engine = create_engine(
    URL(
        account=SNOWFLAKE_ACCOUNT,
        user=SNOWFLAKE_USERNAME,
        warehouse=WAREHOUSE,
        database=DATABASE,
        schema=SCHEMA,
        role=ROLE
        ),
        connect_args={
        'private_key': pkb
        }
)

con = ibis.connect(engine.url)
```
Please let me know if you have any workaround this. 

Thanks.

### Describe the solution you'd like

I would like to have something very simpler way of passing the private key in the `ibis.snowflake.connect()`. for example,
```
con = ibis.snowflake.connect(
    user="user",
    password="password",
    account="safpqpq-sq55555",
    database="IBIS_TESTING/IBIS_TESTING",
    private_key="private_key",
    private_key_password ="private_key_password"
)
```

### What version of ibis are you running?

9.0.0

### What backend(s) are you using, if any?

Snowflake

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by gforsyth** [comment]
Hi @uramith -- we pass all of the arguments to `ibis.snowflake.connect` directly to `snowflake.connector.connect`, so you should be able to just pass the arguments you need directly.

There's also `from_connection` available in Ibis 9.3, which lets you use an existing connection with Ibis, so you could do:

```python
import ibis
import snowflake.connector

conn = snowflake.connector.connect(
    user=SNOWFLAKE_USERNAME,
    account=SNOWFLAKE_ACCOUNT,
    private_key=SNOWFLAKE_PKEY,    # extracted private key from .p8 file
    warehouse=WAREHOUSE,
    database=DATABASE,
    schema=SCHEMA  
)
ibis_con = ibis.snowflake.from_connection(conn)
```

**Comment by uramith** [comment]
Hi @gforsyth thank you! `ibis.snowflake.connect` method worked! Previously, I was using password argument as well along with other args, which was causing authentication error!
